controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -1090,4 +1100,8 @@ public static int getTransactionEpoch(UUID txId) {
     public static long generalizedSegmentId(long segmentId, UUID txId) {
         return computeSegmentId(getSegmentNumber(segmentId), getTransactionEpoch(txId));
     }
+
+    public static int getStartingSegmentNumber(byte[] segmentIndex) {
```

Comment:
Please move this method to the SegmentIndexRecord and keep the logic opaque in this class.

Reply:
Done.